src/DGtal/topology/VoxelComplex.h

```diff
@@ -0,0 +1,621 @@
+
+#if !defined VoxelComplex_h
+/** Prevents repeated inclusion of headers. */
+#define VoxelComplex_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <DGtal/topology/CubicalComplex.h>
+#include <DGtal/topology/Object.h>
+#include <DGtal/kernel/sets/DigitalSetBySTLSet.h>
+#include <DGtal/topology/DigitalTopology.h>
+#include "boost/dynamic_bitset.hpp"
+
+namespace DGtal
+{
+  /**
+  * VoxelComplexCellData stores birth_date and mature
+  * for the persistence thinning algorithm.
+  * @sa CubicalCellData
+  * @sa persistenceAsymetricThinningScheme
+  *
+  * Any cell is stored within a cubical complex with an associated
+  * data. You may define your own associated data but the type must
+  * derive from this class CubicalCellData. Its basic usage is to
+  * store flags associated to the cells, but it may store other
+  * values.
+  *
+  * @note Predefined flags are CubicalComplex::REMOVED,
+  * CubicalComplex::COLLAPSIBLE, CubicalComplex::FIXED,
+  * CubicalComplex::USER1. Other bits can be used to associate an
+  * integer to the cell. The corresponding mask is
+  * CubicalComplex::VALUE.
+  *
+  * @note Such data is notably used in collapse operation
+  * (CubicalComplex::collapse).
+  */
+  struct VoxelComplexCellData: public CubicalCellData {
+    inline VoxelComplexCellData() :
+      CubicalCellData(), birth_date( 0 ){}
+    VoxelComplexCellData( uint32_t d ) :
+      CubicalCellData( d ), birth_date( 0 ) {}
+    VoxelComplexCellData( uint32_t d, uint32_t birth) :
+      CubicalCellData( d ), birth_date( birth ){}
+    uint32_t birth_date;
+  };
+
+  // Forward definitions.
+  template < typename TKSpace, typename TObject, typename TCellContainer >
+  class VoxelComplex;
+  namespace functions {
+    template < typename TKSpace, typename TObject, typename TCellContainer >
+    VoxelComplex< TKSpace, TObject, TCellContainer >&
+    operator-=( VoxelComplex< TKSpace, TObject, TCellContainer >&,
+                const VoxelComplex< TKSpace, TObject, TCellContainer >& );
+    template < typename TKSpace, typename TObject, typename TCellContainer >
+    VoxelComplex< TKSpace, TObject, TCellContainer >
+    operator- ( const VoxelComplex< TKSpace, TObject, TCellContainer >&,
+                const VoxelComplex< TKSpace, TObject, TCellContainer >& );
+
+  } // namespace functions
+  /////////////////////////////////////////////////////////////////////////////
+  // template class VoxelComplex
+  /**
+  * Description of template class 'VoxelComplex' <p> \brief
+  * This class represents a voxel complex living in some
+  * Khalimsky space. Voxel complexes are derived from @sa cubical complexes,
+  * with specialized methods to deal with spels.
+  *
+  * The aim is to implement critical kernels, ie, cliques of spels, as shown
+  * by M.Couprie and G.Bertrand.
+  * References:
+  * - Michel Couprie, Gilles Bertrand. Asymmetric parallel 3D thinning scheme and algorithms based on isthmuses.
```

Comment:
please add the ref in the bibtex and cite it using @cite

Reply:
Added to global.bib and @cited properly.